avocado/plugins/runner_nrunner.py

```diff
@@ -151,9 +152,41 @@ def _get_all_runtime_tasks(test_suite):
                              runnable.uri,
                              None,
                              no_digits)
+            # handles the test task
             task = nrunner.Task(runnable, test_id, status_uris,
                                 nrunner.RUNNERS_REGISTRY_PYTHON_CLASS)
-            runtime_tasks.append(RuntimeTask(task))
+            runtime_task = RuntimeTask(task)
+            runtime_tasks.append(runtime_task)
+
+            # from here, handles the requirements
+            if runnable.requirements is not None:
```

Comment:
nitpick: again, to improve readability, I would suggest invert the order here:

```python
if runnable.requirements is None:
   return runtime_tasks
...
```

Reply:
But I need to process all the runnables before I can return the `runtime_tasks`. The return is out of the `for`.

Reply:
Ohh I missed that, but still maybe you could do this:

```python
if runnable.requirements is None:
   continue
...
...
```

Reply:
I see. The result is the same, but we reduce one level of indentation. I'll change that.